resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -171,8 +171,14 @@ private[spark] class Client(
       yarnClient.init(hadoopConf)
       yarnClient.start()
 
-      logInfo("Requesting a new application from cluster with %d NodeManagers"
-        .format(yarnClient.getYarnClusterMetrics.getNumNodeManagers))
+      try {
+        logInfo(s"Requesting a new application from cluster" +
+          s" ${hadoopConf.get(YarnConfiguration.RM_ADDRESS, YarnConfiguration.DEFAULT_RM_ADDRESS)}" +
+          s" with %d NodeManagers.".format(yarnClient.getYarnClusterMetrics.getNumNodeManagers))
```

Comment:
is it still useful to print the number of node managers?

Reply:
if it can call `yarnClient.getYarnClusterMetrics.getNumNodeManagers` without exception, i tend to keep the original log info.